### ISSUE TITLE
Unable to add auth hooks to config

### ISSUE THREAD

**Issue by samuelwhunter** [supabase/supabase#22031]
Hi there, I'm trying to use the Auth Hooks locally by adding the following snippet found in the docs to the config.toml. 

```
...

[auth.hook.custom_access_token]
enabled = true
uri = "pg-functions://postgres/public/custom_access_token_hook"

...
```

But I'm getting this error message...

`Unknown config fields: [auth.hook.custom_access_token auth.hook.custom_access_token.enabled auth.hook.custom_access_token.uri]`

I've tried deleting the file and running supabase init again. I've also made sure that the CLI is at the latest version (not beta though).

**Comment by encima** [comment]
Can you provide the link to the docs page where you found this?

**Comment by samuelwhunter** [comment]
Of course, yeah sorry. https://supabase.com/docs/guides/auth/auth-hooks#local-development

**Comment by encima** [comment]
Thanks! I am afraid I cannot reproduce this locally, with the latest version (1.148.6) on mac.

Does this occur with a new project or an older one that you re-run supabase init? You could also try removing `supabase/.temp` and running init again.

**Comment by corepay** [comment]
I have same issue, removed temp and reinit fresh toml and added this as same name as my auth hook function:

```
[auth.hook.set_user_owners_app_metadata]
enabled = true
uri = "pg-functions://postgres/public/set_user_owners_app_metadata_hook"
```

supabase -v 1.115.4 on mac

**Comment by encima** [comment]
@corepay please update your CLI and try again

1. Update CLI to latest
2. Remove `supabase/.temp`
3. Run `supabase start`

**Comment by corepay** [comment]
AAAAHHHHH!

So here's the deal - I had supabase installed via brew probably 2 years ago when I was playing around.  I started a new project a few weeks ago and added supabase 1.5.0 to my project and have been building my empire since looks like on the brew version 1.115.4.  Now I know why ther ewere some discrepencies in your docs and what I was seeing - I apologize for the wtf vibes I was sending.

BUT.....couple things FYI

1.  I had problems moving from 115 to 1.5.0.  After upating my brew supabase version, new init and start docker started  downloaded some new stuff - couldnt get going again kept getting 'server unhealthy' and checking my docker only the db showed up in the stack.  I fortunately had a good migration and seeds file so I deleted all my volumes and images and did supa start again everything pulled and started correct.

2.  Now using param in the config.toml I used prior:
```
[auth.hook.set_user_owners_app_metadata]
enabled = true
uri = "pg-functions://postgres/public/set_user_owners_app_metadata_hook"
```
was still receiving the error on supabase stop and start  -> Unknown config fields: ...

I noticed in config.toml this param and stuffed my enabled and uri values there and everything works.
``` 
# This hook runs before a token is issued and allows you to add additional claims based on the authentication method used.
[auth.hook.custom_access_token]
```

I bring this up becuase the docs [here](https://supabase.com/docs/guides/auth/auth-hooks#local-development) on this  suggest that I can name the hook whatever I wanted `Modify the auth.hook.<hook_name> field...` but it looks like we can only modify the `auth.hook.custom_access_token` field.  

![image](https://github.com/supabase/supabase/assets/48992923/1c497376-70a6-4a09-abaa-0e77550c38d8)

**Comment by encima** [comment]
thanks for the detailed feedback also! The docs may be hard to parse here so it could be clearer. 
To clarify: `<hook_name>` means any name that matches to one of the Auth hooks, so:
- `custom_access_token`
- `mfa_verification_attempt`
- `password_verification_attempt`

The hooks are listed [here](https://supabase.com/docs/guides/auth/auth-hooks). Will close this one out!

**Comment by vood** [comment]
The <hook_name> being a predefined list of hook names isn't clear. I only was able to figure this out by finding this issue. I think the doc need to be updated for clarity. I actually thought that hook name can be anything so that multiple hooks can be provided with different names
